src/main/java/com/nutomic/syncthingandroid/activities/DeviceActivity.java

```diff
@@ -82,11 +90,21 @@
 
     private boolean mDeviceNeedsToUpdate;
 
+    private boolean mIsShowingDiscardDialog;
+    private boolean mIsShowingCompressionDialog;
+    private boolean mIsShowingDeleteDialog;
```

Comment:
I suppose you could get rid of these variables, and just check if the dialogs are null?